exercises/pascals-triangle/src/example.c

```diff
@@ -0,0 +1,26 @@
+#include "pascals_triangle.h"
+
+long long unsigned int factorial(int n)
+{
+   return (n == 0) ? 1 : n * factorial(n - 1);
+}
+
+long long unsigned int combination(int n, int r)
+{
+   return (r > n) ? 0 : factorial(n) / (factorial(n - r) * factorial(r));
+}
+
+int pascals_triangle(int count, int input[][((count < 0) ? 1 : count)])
+{
+   if (count < 0) {
+      memset(input, -1, sizeof(int) * count * count);
+      return -1;
+   }
+   memset(input, 0, sizeof(int) * count * count);
```

Comment:
what I mean is that on a good compiler, the code as is, will give a warning - thus it should change. Here you have not used a single row as you say you intend, you have used however much is `count` and hoped that it is never less than `-1`, if a user or test were to enter `count = -20`, this would attempt to set 400 rows! (`count * count == -20 * -20 == 400`)